ksql-streams/src/main/java/io/confluent/ksql/execution/streams/ExecutionStepFactory.java

```diff
@@ -140,6 +143,25 @@ public static StreamSource streamSource(
     );
   }
 
+  public static <K> StreamFlatMap<K> streamFlatMap(
+      final QueryContext.Stacker stacker,
+      final ExecutionStep<KStreamHolder<K>> source,
+      final LogicalSchema resultSchema,
+      final List<FunctionCall> functionCalls,
+      final FunctionRegistry functionRegistry,
+      final LogicalSchema inputSchema
+  ) {
+    final QueryContext queryContext = stacker.getQueryContext();
+    return new StreamFlatMap<>(
+        new DefaultExecutionStepProperties(resultSchema, queryContext),
+        source,
+        functionCalls,
+        functionRegistry,
+        inputSchema,
+        resultSchema
```

Comment:
Does it make sense to pass the `resultSchema` in twice? i.e. once in `DefaultExecutionStepProperties` and again here.

`StreamFlatMap.getOutputSchema` doesn't seem to be called.

Reply:
changed